platform/broadcom/saibcm-modules/debian/opennsl-modules.init

```diff
@@ -32,14 +32,16 @@ function create_devices()
 #                                 level logs
 function load_kernel_modules()
 {
-    modprobe linux-kernel-bde dmasize=32M maxpayload=128 debug=4 dma_debug=1
+    modprobe linux-kernel-bde dmasize=128M maxpayload=128 debug=4 dma_debug=1 usemsi=1
     modprobe linux-user-bde
+    modprobe psample
```

Comment:
in linux kernel, there is another psample kernel module? How do you ensure you are installing the correct one?

admin@vlab-01:/lib/modules$ sudo find . -name "psample*"
./4.9.0-9-2-amd64/kernel/net/psample
./4.9.0-9-2-amd64/kernel/net/psample/psample.ko

Reply:
Yes, I am working on this still, I do see two modules one from Linux kernel and another from bcm gpl modules. Shall fix this soon.

Reply:
Did some study on this, there are two module files created as below.
lib/modules/4.9.0-9-2-amd64/extra/psample.ko
lib/modules/4.9.0-9-2-amd64/kernel/net/psample/psample.ko

But the modules.dep is having the psample.ko module build and installed in "/extra" directory which is from platform/Broadcom/saibcm-modules as below.

**<modules.dep>**
extra/linux-knet-cb.ko: extra/linux-bcm-knet.ko extra/linux-kernel-bde.ko extra/psample.ko
extra/psample.ko:

Reply:
the question is not answer, when you call modprobe psample, how does the kernel know which psample driver to installer? the vanilla one or the bcm one.

Reply:
By default, modules will load based on what is present in the **modules.dep** file. The modules.dep file in SONIC box has **only bcm** psample.ko file.

Anyways to make sure we load the correct module I have replaced the "modprobe psample" **with** insmod /lib/modules/$(uname -r)/extra/psample.ko.